client/state/reader/site-blocks/reducer.js

```diff
@@ -40,6 +41,25 @@ export const items = createReducer(
 				[ action.payload.ID ]: true,
 			};
 		},
+		[ READER_SITE_BLOCKS_RECEIVE ]: ( state, action ) => {
+			if ( ! action.payload || ! action.payload.sites ) {
+				return state;
+			}
+
+			const newBlocks = reduce(
+				action.payload.sites,
+				( obj, site ) => {
+					obj[ site.ID ] = true;
+					return obj;
+				},
+				{}
+			);
+
+			return {
+				...state,
```

Comment:
Hm, are those blocks be removable? If yes, how are we handling the consequences from a block removal here ? 

To illustrate the case better, let's imagine we have a block in the state that has already been deleted. I expect it won't be returned from the endpoint, but with the current reducer it'll still be kept after the receive action is dispatched. I'm not sure that's the expected behavior.

Reply:
So to sum up:

1. The state looks like this:

```123: true```

2. The site block on 123 is removed outside of Calypso.
3. The `READER_SITE_BLOCKS_RECEIVE` action doesn't tell us anything about site 123, because it's not blocked.

We might receive multiple pages of `READER_SITE_BLOCKS_RECEIVE` for each user, so we can't discard the previous state each time.  

Tricky! I'll have a ponder on that one.